File: RL/Part-I/src/Action.py
class Action:
    def __init__(self, ind, name):
        self.index = ind
        self.name = name

    def __repr__(self):
        return "Index: " + str(self.index) + " Name: " + str(self.name)

    def __eq__(self, other):
        if isinstance(other, self.__class__):
            return self.index == other.index and self.name==other.name
        return False

File: EM and NLP/src/main/java/Action.java
import java.io.Serializable;

/**
 * Created by tarun on 4/6/17.
 */
public class Action implements Serializable {
    int index;
    int gotox;
    String name;

    public Action(int ind, String name) {
        this(ind, name, -1);
    }

    public Action(int ind, String name, int gotox) {
        this.index = ind;
        this.name = name;
        this.gotox = gotox;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Action)) {
            return false;
        }
        Action other = (Action)o;

        return (other.index==this.index && other.name.equals(this.name) && other.gotox==this.gotox);
    }

    @Override
    public String toString() {
        return "Index: " + this.index + " Name: " + this.name + " Goto: " + this.gotox;
    }
}
